proto/desmos/posts/v1/models.proto

```diff
@@ -0,0 +1,258 @@
+syntax = "proto3";
+package desmos.posts.v1;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/any.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/desmos-labs/desmos/v3/x/posts/types";
+
+// Post contains all the information about a single post
+message Post {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
+  // Id of the subspace inside which the post has been created
+  uint64 subspace_id = 1 [ (gogoproto.customname) = "SubspaceID" ];
+
+  // Unique id of the post
+  uint64 id = 2 [ (gogoproto.customname) = "ID" ];
+
+  // (optional) External id for this post
+  string external_id = 3 [ (gogoproto.customname) = "ExternalID" ];
+
+  // (optional) Text of the post
+  string text = 4;
+
+  // (optional) Entities connected to this post
+  Entities entities = 5;
+
+  // Author of the post
+  string author = 6;
+
+  // (optional) Id of the original post of the conversation
+  uint64 conversation_id = 7 [ (gogoproto.customname) = "ConversationID" ];
+
+  // A list this posts references (either as a reply, repost or quote)
+  repeated PostReference referenced_posts = 8 [ (gogoproto.nullable) = false ];
+
+  // Reply settings of this post
+  ReplySetting reply_settings = 9;
+
+  // Creation date of the post
+  google.protobuf.Timestamp creation_date = 10
+      [ (gogoproto.nullable) = false, (gogoproto.stdtime) = true ];
+
+  // (optional) Last edited time of the post
+  google.protobuf.Timestamp last_edited_date = 11
+      [ (gogoproto.stdtime) = true ];
+}
+
+// PostReference contains the details of a post reference
+// TODO: Move this inside the Post object
```

Comment:
I thought of moving it to be a inner type because the only place where the `PostReference` type is used is within the `Post` type itself. There are no other ways, so it doesn't make much sense to leave it where it is right now. Also, the current naming contains the unwanted `Post` prefix which could then be deleted (the go type would become `Post_Reference`)

Reply:
Ended up simply deleting this TODO as I changed my mind. Going to stick to the ADR specification as the model is used elsewhere as well (eg. inside `MsgCreatePost`)